### PR TITLE
Fix the dreaded legacy build

### PR DESCRIPTION
**Pull request by magnumripper**
From the looks of it it was b0rken again since some time.

### CODE REVIEW

File: src/formats.c [hunk 1]
```diff
@@ -72,8 +72,10 @@ static void test_fmt_case(struct fmt_main *format, void *binary,
  */
 int fmt_match(const char *req_format, struct fmt_main *format, int override_disable)
 {
+#ifndef BENCH_BUILD
 	int enabled = override_disable || !((options.flags & FLG_TEST_CHK) || options.listconf) ||
 		!cfg_get_bool(SECTION_DISABLED, SUBSECTION_FORMATS, format->params.label, 0);
+#endif
```
Comment (solardiz) [inline src/formats.c]: This could simply be moved under the existing `#ifndef` below, near the `char *pos = NULL;` line.

**Review by solardiz — APPROVED** [review]

File: src/formats.c (hunk 1 above)
Comment (magnumripper) [reply src/formats.c]: Right, I'll do that. I'm so used with trying to keep differences from core branch's code at a graspable level but we can't merge that branch anymore anyway due to git-cvs having eroded to being unusable with reasonable effort.

Oh btw I have a semaphore in my head shouting we have some tiny(?) rule change from core not yet merged into Jumbo

Comment (magnumripper) [reply src/formats.c]: Right, I'll do so. I'm still used to try keeping core files close enough to CVS that I can understand the diffs but now that `git-cvs` corroded to completely unusable without unreasonable effort, it's of no use anymore.

Comment (magnumripper) [reply src/formats.c]: I see that's not even a core function btw... A whole lot of them aren't 😆

Comment (magnumripper) [reply src/formats.c]: No, wait - it is. It's the `bench.c` one that's completely new code (and an awful cludge at that)

**Comment by magnumripper** [comment]
Perhaps I was a spoiled child but this mandatory Travis check of an as-trivial-as-it-gets commit has been going on for over 4h now even though most parts of the globe are off to relax somewhere and not committing anything at all. Are there better alternatives? Or should we simply throw some money at them?

**Comment by ghost** [comment]
> Perhaps I was a spoiled child but this mandatory Travis check of an as-trivial-as-it-gets commit has been going on for over 4h now even though most parts of the globe are off to relax somewhere and not committing anything at all. Are there better alternatives? Or should we simply throw some money at them?

For weeks, Travis.org is taking 6, 8, 10 hours to build. Yes, 80% of it is "waiting time". They have 4, maybe 5 hundred bots, but they have almost 6 thousand jobs in their backlog NOW (Saturday).

**[edited]**
* My latest run (`#2004 passed` Ran for 12 hrs 2 sec)

**Comment by solardiz** [comment]
> For weeks, Travis.org is taking 6, 8, 10 hours to build.

FWIW, I setup Travis (I think it's .com) for LKRG, and it consistently completes (trivial) builds in minutes (we're only testing being able to build LKRG now - not yet testing that the built LKRG actually works).

**Comment by magnumripper** [comment]
I see now Travis actually finished successfully 20 hours ago, but Github is still "waiting for status to be reported".

@solardiz just merge this please!

**Comment by magnumripper** [comment]
Perhaps the finished one was older. I now found "*Builds have been temporarily disabled for public repositories due to a negative credit balance. Please go to the Plan page to replenish your credit balance or alter your Consume paid credits for OSS setting.*"

https://travis-ci.com/github/openwall/john

**Comment by solardiz** [comment]
> _Builds have been temporarily disabled for public repositories due to a negative credit balance. Please go to the Plan page to replenish your credit balance or alter your Consume paid credits for OSS setting._

Clicking either of the two links in that message results in:

```
There was an error, please try again.
```

Let's see if things start working later.

That said, I think our use of CI is quite wasteful, e.g. when we ran many builds and tests of those builds on recent introduction of rulesets and force-pushes to that PR. Can we find a clean way to skip most CI tests (perhaps except for the quickest, which involves basic loading of `john.conf`) on such PRs?

**Comment by ghost** [comment]
> That said, I think our use of CI is quite wasteful, e.g. when we ran many builds and tests of those builds on recent introduction of rulesets and force-pushes to that PR. Can we find a clean way to skip most CI tests (perhaps except for the quickest, which involves basic loading of john.conf) on such PRs?

There are some ways to do so. Run some builds only in "main" branch, use the `[skip ci]` in the commit message, ...

I remember I used `[skip ci]` some time ago. Then, only magnum was able to merge the PR. If we run some jobs only in "main" branch, we will test only **after** merge.

**Pull request merged**